pkg/manifests/manifests.go

```diff
@@ -78,17 +79,26 @@ func (f *Factory) RouterDaemonSet(cr *ingressv1alpha1.ClusterIngress) (*appsv1.D
 
 	ds.Name = name
 
-	ds.Labels = map[string]string{
-		"app": "router",
-	}
+	ds.Spec.Template.Labels["router"] = name
 
 	ds.Spec.Selector = &metav1.LabelSelector{
 		MatchLabels: map[string]string{
-			"app": name,
+			"app":    "router",
+			"router": name,
 		},
 	}
 
-	ds.Spec.Template.Labels["app"] = name
+	routeSelector, err := metav1.LabelSelectorAsSelector(cr.Spec.RouteSelector)
+	if err != nil {
+		return nil, fmt.Errorf("clusteringress %q has invalid spec.routeSelector: %v", cr.Name, err)
+	}
+	env := []corev1.EnvVar{
+		{Name: "ROUTER_SERVICE_NAME", Value: cr.Name},
+		{Name: "ROUTER_SERVICE_NAMESPACE", Value: ds.Namespace},
```

Comment:
Here the use of external assets makes the code confusing—it isn't obvious where `ds.Namespace` is set, or whether an assignment was omitted through oversight.  I think the code would be clearer with an explicit assignment `namespace := ds.Namespace` or (inclusive) comment that we are deliberately using the namespace that is hard-coded in the asset.  Thoughts?

Alternatively, since you are appending `env` to `ds.Spec.Template.Spec.Containers[0].Env`, you could just as well hard-code the environment variable in the yaml.

Reply:
Do you think `ROUTER_SERVICE_NAME` makes more sense equal to ClusterIngress.Name or DaemonSet.Name? I'm not sure I have a strong argument either way yet.

Settings the namespace in the static asset seems okay. The confusion between the external assets and code is unfortunately inevitable, so any suggestions for keeping it sane are welcome.

Reply:
I also went ahead and moved the service namespace to YAML since it's static